### DISCUSSION TITLE
rTorrent Daemon

### DISCUSSION THREAD

**Discussion by ajhorvath20** [cross-seed/cross-seed#114] [Q&A] [answered]
Hi, I have `daemon` mode working perfectly fine with `qbittorrent` but I am not sure how to use `daemon` in `rtorrent`. 

Do I need to change something in `rtorrent.rc`?

Thanks

**Comment by mmgoodnow** [comment, marked as answer]
I have a shell script called rtorrent-postprocess.sh. The rtorrent scripting language is incredibly poorly documented and hard to use in general, so I like to pass all the info into the shell script and then do all the actual work there (which still isn't much)

```
method.set_key=event.download.finished,filebot,"execute={/config/rtorrent/rtorrent-postprocess.sh,$d.base_path=,$d.name=,$d.custom1=}"
```

The shell script gets called with the base path, name, and label as arguments. 

Inside the shell script, we do 
```
curl -XPOST http://cross-seed:2468/api/webhook --data-urlencode "name=$2"
```

If you use Docker, this still happens inside your docker container so curl must be installed.

**Reply by wbm37** [reply]
Hi, thanks for this helpful reply, but I am having a problem.

I setup the docker container and it seems to curl and get the name just fine, but for every time it finishes a torrent, it shows this in the cross-seed container logs.

```
info: [server] Received  { name: 'REDACTED', trackers: 'REDACTED', outputDir: '/cross-seed' }
error: could not find a torrent with the criteria [Object: null prototype] {
  name: 'REDACTED',
  trackers: 'REDACTED',
  outputDir: '/cross-seed'
}
```
It gets the name correct, but throws the error and doesn't go beyond that. 

If I just put into the script

`curl -XPOST http://cross-seed:2468/api/webhook --data-urlencode "name=$2"`

I get

```
info: [server] Received  { name: 'REDACTED' }
info: [server] Did not search for { name: 'REDACTED' }
```

It runs fine if I exec into the container and run

`cross-seed search`

Kinda at whits end with this and can't quite pinpoint the problem.

**Reply by mmgoodnow** [reply]
Hmmm, I may have introduced a bug recently. I'll look into it!

**Reply by mmgoodnow** [reply]
I looked into this and it was indeed a bug, fixed in the latest version. However I do have to warn you that the documentation is out of date and putting ` trackers: 'REDACTED', outputDir: '/cross-seed' ` isn't actually supported at the moment, so don't be surprised when it doesn't affect anything.

**Reply by wbm37** [reply]
Good to hear! I really appreciate you looking into it. And that's okay, I'll remove it for now, thanks for letting me know. 

Have a good rest of your day!

**Reply by mmgoodnow** [reply]
@Wmartin417 `trackers: 'REDACTED', outputDir: '/cross-seed'` should be back to working as specified in the docs!

**Reply by keldian** [reply]
> I have a shell script called rtorrent-postprocess.sh. The rtorrent scripting language is incredibly poorly documented and hard to use in general, so I like to pass all the info into the shell script and then do all the actual work there (which still isn't much)
> 
> ```
> method.set_key=event.download.finished,filebot,"execute={/config/rtorrent/rtorrent-postprocess.sh,$d.base_path=,$d.name=,$d.custom1=}"
> ```
> 
> The shell script gets called with the base path, name, and label as arguments.
> 
> Inside the shell script, we do
> 
> ```
> curl -XPOST http://cross-seed:2468/api/webhook --data-urlencode "name=$2"
> ```
> 
> If you use Docker, this still happens inside your docker container so curl must be installed.

Could you be more explicit on how to set this up? Is the script exactly [this one](https://raw.githubusercontent.com/filebot/plugins/master/bash/rtorrent-postprocess.sh), with just `curl -XPOST http://cross-seed:2468/api/webhook --data-urlencode "name=$2"` copy&pasted on a new line?

**Reply by mmgoodnow** [reply]
Yeah pretty much. You can try it out from the command line first if you map 2468, which will help you debug

**Reply by keldian** [reply]
I'm running rtorrent from a docker image. If I understand correctly, I'd have to test that from the rtorrent container's shell, and that container does not have filebot installed, so I don't believe the script would work. Not sure how to work around this–installing rtorrent directly on the host is out of the question.

Yep, confirmed:
```
root@rutorrent:/config/rtorrent# sh rtorrent-postprocess.sh
rtorrent-postprocess.sh: line 11: filebot: not found
```

**Reply by mmgoodnow** [reply]
if you don't have filebot then delete the filebot part.

**Reply by keldian** [reply]
```
Configuration file already exists.
info: Validating your configuration...
info: Your configuration is valid!
info: [server] Server is running on port 2468, ^C to stop.
info: [server] Received  { name: '' }
info: [server] Found 0 torrents for { name: '' }
```
Seems to work. By the way, I didn't have to map the port–I'm guessing the cross-seed image already took care of that on its own.

Thank you.
